src/griddyn/loads/aggregateLoad.cpp

```diff
@@ -114,14 +114,16 @@ void aggregateLoad::pFlowObjectInitializeA (coreTime time0, std::uint32_t flags)
         // do a first pass of loading
         double rem = 1.0;
 
-        setP (((aggregateLoad *)sLoad)->getP ());  // so this composite load function has direct access to the
+        aggregateLoad* sLoadRepr = dynamic_cast<aggregateLoad*>(sLoad);
```

Comment:
this is a bit of weird code.  You need to use reinterpret_cast here,  dynamic_cast won't work properly.  We know the object is a zipload, but for the compiler to allow these calls which are purely part of the zipload interface it has to think it is a aggregate load.  
We probably should find a different way of doing this but for now it needs to be reinterpret_cast not dynamic_cast